# Large-language-models-Protein


**Large language models generate functional protein sequences across diverse familie**

Madani A, Krause B, Greene E R, et al. Large language models generate functional protein sequences across diverse families[J]. Nature Biotechnology, 2023: 1-8.

https://www.nature.com/articles/s41587-022-01618-2

## Language models of protein sequences at the scaleof evolution enable accurate structure prediction

Lin Z, Akin H, Rao R, et al. Language models of protein sequences at the scale of evolution enable accurate structure prediction[J]. BioRxiv, 2022, 2022: 500902.

https://www.biorxiv.org/content/10.1101/2022.07.20.500902v1.full.pdf?utm_campaign=M2D2%20Community%20Round-Up&utm_medium=email&utm_source=Revue%20newsletter

## Papers on machine learning for proteins

### Background

We recently released a [review](https://arxiv.org/abs/1811.10775) of machine learning methods in protein engineering, but the field changes so fast and there are so many new papers that any static document will inevitably be missing important work. This format also allows us to broaden the scope beyond engineering-specific applications. We hope that this will be a useful resource for people interested in the field.

To the best of our knowledge, this is the first public, collaborative list of machine learning papers on protein applications. We try to classify papers based on a combination of their applications and model type. If you have suggestions for other papers or categories, please make a pull request or issue!

### Format

Within each category, papers are listed in reverse chronological order (newest first). Where possible, a link should be provided.

### Categories

[Reviews](#reviews)  
[Tools and datasets](#tools-and-datasets)  
[Machine-learning guided directed evolution](#machine-learning-guided-directed-evolution)  
[Representation learning](#representation-learning)  
[Unsupervised variant prediction](#unsupervised-variant-prediction)  
[Generative models](#generative-models)  
[Biophysics](#biophysics)
[Predicting stability](#predicting-stability)  
[Predicting structure from sequence](#predicting-structure-from-sequence)  
[Predicting sequence from structure](#predicting-sequence-from-structure)  
[Classification, annotation, search, and alignments](#classification-annotation-search-and-alignments)  
[Predicting interactions with other molecules](#predicting-interactions-with-other-molecules)  
[Other supervised learning](#other-supervised-learning)

### Reviews

**From sequence to function through structure: deep learning for protein design.**  
Noelia Ferruz, Michael Heinzinger, Mehmet Akdel, Alexander Goncearenco, Luca Naef, Christian Dallago.  
*Preprint, September 2022.*  
[[10.1101/2022.08.31.505981](https://doi.org/10.1101/2022.08.31.505981)]

### Tools and datasets

**PDBench: Evaluating Computational Methods for Protein-Sequence Design**
Leonardo V Castorina, Rokas Petrenas, Kartic Subr, Christopher W Wood.
*Bioinformatics, January 2023.*  
[[10.1093/bioinformatics/btad027](https://doi.org/10.1093/bioinformatics/btad027)]]


### Representation learning

**Codon language embeddings provide strong signals for protein engineering.**  
Carlos Outeiral, Charlotte M. Deane.  
*Preprint, December 2022.*  
[[10.1101/2022.12.15.519894](https://doi.org/10.1101/2022.12.15.519894)]


### Generative models

**De novo design of high-affinity protein binders to bioactive helical peptides.**  
Susana Vázquez Torres, Philip J. Y. Leung, Isaac D. Lutz, Preetham Venkatesh, Joseph L. Watson, Fabian Hink, Huu-Hien Huynh, Andy Hsien-Wei Yeh, David Juergens, Nathaniel R. Bennett, Andrew N. Hoofnagle, Eric Huang, Michael J MacCoss, Marc Expòsit, Gyu Rie Lee, Paul M. Levine, Xinting Li, Mila Lamb, Elif Nihal Korkmaz, Jeff Nivala, Lance Stewart, Joseph M. Rogers, David Baker.  
*Preprint, December 2022.*  
[[10.1101/2022.12.10.519862](https://doi.org/10.1101/2022.12.10.519862)]

### Biophysics

**Machine Learning Coarse-Grained Potentials of Protein Thermodynamics.**  
Maciej Majewski, Adrià Pérez, Philipp Thölke, Stefan Doerr, Nicholas E. Charron, Toni Giorgino, Brooke E. Husic, Cecilia Clementi, Frank Noé, Gianni De Fabritiis.  
*Preprint, December 2022.*  
[[arxiv](https://arxiv.org/abs/2212.07492)]


### Predicting structure from sequence

**AFsample: Improving Multimer Prediction with AlphaFold using Aggressive Sampling.**  
Björn Wallner.  
*Preprint, December 2022.*  
[[10.1101/2022.12.20.521205](https://doi.org/10.1101/2022.12.20.521205)]


### Predicting sequence from structure

**De novo protein design by inversion of the AlphaFold structure prediction network.**  
Casper Goverde, Benedict Wolf, Hamed Khakzad, Stéphane Rosset, Bruno E. Correia.  
*Preprint, December 2022.*  
[[10.1101/2022.12.13.520346](https://doi.org/10.1101/2022.12.13.520346)]



### Classification, annotation, search, and alignments

**Fast and accurate protein function prediction from sequence through pretrained language model and homology-based label diffusion.**  
Qianmu Yuan, Junjie Xie, Jiancong Xie, Huiying Zhao, Yuedong Yang
*Preprint, December 2022.*  
[[10.1101/2022.12.05.519119](https://doi.org/10.1101/2022.12.05.519119)]


### Predicting interactions with other molecules

**DiffBP: Generative Diffusion of 3D Molecules for Target Protein Binding.**  
Haitao Lin, Yufei Huang, Meng Liu, Xuanjing Li, Shuiwang Ji, Stan Z. Li.  
*Preprint, November 2022.*  
[[arxiv](https://arxiv.org/abs/2211.11214)]


### Other supervised learning


**Turnover number predictions for kinetically uncharacterized enzymes using machine and deep learning.**  
Alexander Kroll, Xiao-Pan Hu, Nina A. Liebrand, Martin J. Lercher.  
*Preprint, November 2022.*  
[[10.1101/2022.11.10.516024](https://doi.org/10.1101/2022.11.10.516024)]
