# Papers on Large-language-models-Protein
### Background

We recently released a [review](https://arxiv.org/abs/1811.10775) of machine learning methods in protein engineering, but the field changes so fast and there are so many new papers that any static document will inevitably be missing important work. This format also allows us to broaden the scope beyond engineering-specific applications. We hope that this will be a useful resource for people interested in the field.

To the best of our knowledge, this is the first public, collaborative list of machine learning papers on protein applications. We try to classify papers based on a combination of their applications and model type. If you have suggestions for other papers or categories, please make a pull request or issue!

### Format

Within each category, papers are listed in reverse chronological order (newest first). Where possible, a link should be provided.

### Categories

[Reviews](#reviews)  
[Tools and datasets](#tools-and-datasets)  
[Machine-learning guided directed evolution](#machine-learning-guided-directed-evolution)  
[Representation learning](#representation-learning)  
[Unsupervised variant prediction](#unsupervised-variant-prediction)  
[Generative models](#generative-models)  
[Biophysics](#biophysics)
[Predicting stability](#predicting-stability)  
[Predicting structure from sequence](#predicting-structure-from-sequence)  
[Predicting sequence from structure](#predicting-sequence-from-structure)  
[Classification, annotation, search, and alignments](#classification-annotation-search-and-alignments)  
[Predicting interactions with other molecules](#predicting-interactions-with-other-molecules)  
[Other supervised learning](#other-supervised-learning)

### Reviews   
**Large language models generate functional protein sequences across diverse families**    
Ali Madani, Ben Krause, Eric R. Greene, Subu Subramanian, Benjamin P. Mohr, James M. Holton, Jose Luis Olmos Jr., Caiming Xiong, Zachary Z. Sun, Richard Socher, James S. Fraser & Nikhil Naik     
*Published online: 26 January 2023*    
[[s41587-022-01618-2](https://www.nature.com/articles/s41587-022-01618-2)]


**Language models of protein sequences at the scale of evolution enable accurate structure prediction**     
Zeming Lin, Halil Akin, Roshan Rao, Brian Hie, Zhongkai Zhu, Wenting Lu, Allan dos Santos Costa, Maryam Fazel-Zarandi, Tom Sercu, Sal Candido, Alexander Rives    
*Published online: 16 Mar 2023*   
[[10.1101/2022.07.20.500902]([https://doi.org/10.1101/2022.07.20.500902](https://www.biorxiv.org/content/10.1101/2022.07.20.500902v1.full.pdf?utm_campaign=M2D2%20Community%20Round-Up&utm_medium=email&utm_source=Revue%20newsletter))] 


**Controllable protein design with language models**   
Noelia Ferruz , Birte Höcker   
*Published: 22 June 2022*    
[[s42256-022-00499-z](https://www.nature.com/articles/s42256-022-00499-z)]   

**Transformer protein language models are unsupervised structure learners**
Roshan Rao, Joshua Meier, Tom Sercu,Sergey Ovchinnikov, Alexander Rives     
*Posted： December 15, 2020.*
[[10.1101/2020.12.15.422761](https://www.biorxiv.org/content/10.1101/2020.12.15.422761v1.full)]   


**ProtGPT2 is a deep unsupervised language model for protein design**  
Noelia Ferruz, Steffen Schmidt, Birte Höcker   
*Published: 27 July 2022*
[[s41467-022-32007-7](https://www.nature.com/articles/s41467-022-32007-7)]   dt


**123**


### Tools and datasets
### Representation learning
### Generative models
### Biophysics
### Predicting structure from sequence
### Predicting sequence from structure
### Classification, annotation, search, and alignments
### Predicting interactions with other molecules
### Other supervised learning
