# LLM-based Protein Papers


>What can **Large Language Models (LLMs)** do for Protein? 

üôå This repository collects some LLM-based protein papers.

üòé Welcome to recommend missing papers through **`Adding Issues`** or **`Pull Requests`**. 




## üîî News
- **2023-07  We create this repository to maintain a paper list on *Large Language Models* appiled in *Protein*.**


*Todo:*
1. - [ ] `Fine-grained classification of papers`
2. - [ ] `Update paper project / code`

   
## Content


  
- [üìú Papers](#papers)
  - [üß¨ Generation of protein sequence](#generation-of-protein-sequence)
  - [üîé Prediction of protein function](#prediction-of-protein-function)


---

##  Papers

###  Generation of protein sequence
- \[[bioRxiv](https://doi.org/10.1101/2023.06.22.546080)\]  
  DeepRank-GNN-esm: A Graph Neural Network for Scoring Protein-Protein Models using Protein Language Model`2023.06`\[[DeepRank](https://github.com/DeepRank/DeepRank-GNN-esm)\]

  
- \[[bioRxiv](https://doi.org/10.1101/2023.04.24.538081 )\]  
  RaptorX-Single: single-sequence protein structure prediction by integrating protein language models`2023.06`
- \[[nature biotechnology](https://www.nature.com/articles/s41587-022-01618-2)\]  
  Large language models generate functional protein sequences across diverse families `2023.01`
- \[[nature biotechnology](https://www.nature.com/articles/s41587-022-01518-5)\]  
  De novo protein design with a language model`2022.10`
- \[[nature communications](https://www.nature.com/articles/s41467-022-32007-7)\]  
  ProtGPT2 is a deep unsupervised language model for protein design`2022.07`
- \[[nature machine intelligence](https://www.nature.com/articles/s42256-022-00499-z)\]  
  Controllable protein design with language models `2022.06`
- \[[bioRxiv]( https://doi.org/10.1101/2021.07.18.452833)\]   
  Deep neural language modeling enables functional protein generation across families`2021.07¬∑
- \[[bioRxiv](https://www.biorxiv.org/content/10.1101/2020.12.15.422761v1.full)\]  
  Transformer protein language models are unsupervised structure learners`2020.12`
- \[[arXiv](https://arxiv.org/abs/2006.03555)\]   
  Masked Language Modeling for Proteins via Linearly Scalable Long-Context Transformers`2020.10`
- \[[arXiv](https://doi.org/10.48550/arXiv.2004.03497)\]  
  ProGen: Language Modeling for Protein Generation`2020.03`

###  Prediction of protein function

- \[[ScienceDirect](https://doi.org/10.1016/j.sbi.2023.102641)\]   
  Finding functional motifs in protein sequences with deep learning and natural language models`2023.08`

  
- \[[arXiv](https://arxiv.org/abs/2306.15912)\]   
  Protein-DNA binding sites prediction based on pre-trained protein language model and contrastive learning`2023.06`
- \[[Briefings in Bioinformatics](https://doi.org/10.1093/bib/bbad217)\]  
  Improved structure-related prediction for insufficient homologous proteins using MSA enhancement and pre-trained language model`2023.06`
- \[[Briefings in Bioinformatics](https://doi.org/10.1093/bib/bbad173)\]  
  Fast and accurate protein intrinsic disorder prediction by using a pretrained language model`2023.05 `\[[LMDisorder](https://github.com/biomed-AI/LMDisorder)\]
- \[[Research Square](https://doi.org/10.21203/rs.3.rs-2852098/v1)\]   
   Large language models improve annotation of viral proteins`2023.05`
- \[[scientific reports](https://www.nature.com/articles/s41598-023-31612-w)\]  
  Graph-BERT and language model-based framework for protein‚Äìprotein interaction identification`2023.04`
- \[[ScienceDirect](https://doi.org/10.1016/j.gpb.2023.04.001)\]  
  NetGO 3.0: Protein Language Model Improves Large-scale Functional Annotations`2023.04`
- \[[Science](https://www.science.org/doi/10.1126/science.ade2574)\]  
  EvolutioResearch Squarenary-scale prediction of atomic-level protein structure with a language model`2023.03`
- \[[bioRxiv](https://doi.org/10.1101/2022.07.20.500902)]   
  Language models of protein sequences at the scale of evolution enable accurate structure prediction `2023.03`
- \[[IEEE](https://ieeexplore.ieee.org/abstract/document/10030025)\]   
  Sequence-Structure Embeddings via Protein Language Models Improve on Prediction Tasks`2022.12`
- \[[plos computational biology](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010793)\]   
  Integrating unsupervised language model with triplet neural networks for protein gene ontology prediction`2022.12`
- \[[scientific reports](https://www.nature.com/articles/s41598-022-21366-2)\]  
  Improving protein succinylation sites prediction using embeddings from protein language model`2022.10`
- \[[plos computational biology](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010668)\]  
  DMFpred: Predicting protein disorder molecular functions based on protein cubic language model`2022.10`
- \[[MDPI](https://doi.org/10.3390/axioms11090469)\]  
  PTG-PLM: Predicting Post-Translational Glycosylation and Glycation Sites Using Protein Language Models and Deep Learning`2022.09`
- \[[Cell Systems](https://www.cell.com/cell-systems/pdf/S2405-4712(22)00038-2.pdf)\]  
  Evolutionary velocity with protein language models predicts evolutionary dynamics of diverse proteins`2022.04`
- \[[nature machine intelligence](https://www.nature.com/articles/s42256-022-00457-9)\]  
  Learning functional properties of proteins with language models`2022.03`
- \[[NAR](https://doi.org/10.1093/nargab/lqac004)\]  
  PANDA2: protein function prediction using graph neural networks`2022.02`\[[PANDA2](http://dna.cs.miami.edu/PANDA2/)\]
- \[[SPRINGER LINK](https://link.springer.com/article/10.1007/s00439-021-02411-y)\]  
  Embeddings from protein language models predict conservation and variant effects`2021.12`
- \[[arXiv](https://arxiv.org/abs/2110.15527)\]   
  Pre-training Co-evolutionary Protein Representation via A Pairwise Masked Language Model`2021.10`
- \[[Cell SystemS](https://www.cell.com/cell-systems/pdf/S2405-4712(21)00203-9.pdf)\]  
  Learning the protein language:Evolution, structure, and function`2021.05`
- \[[arXiv](https://doi.org/10.48550/arXiv.2006.15222)\]  
  BERTology Meets Biology: Interpreting Attention in Protein Language `2021.03`
- \[[MDPI](https://doi.org/10.3390/cells8020122)\]  
  A High Efficient Biological Language Model for Predicting Protein‚ÄìProtein Interactions`2019.02`

## Contribution
### üë• Contributors
### üéâ Contributing ( welcome ! )

- ‚ú® Add a new paper or update an existing Protein-related LLM paper.
- üßê Use the same format as existing entries to describe the work.
- üòÑ A very brief explanation why you think a paper should be added or updated is recommended (Not Neccessary) via **`Adding Issues`** or **`Pull Requests`**.

**Don't worry if you put something wrong, they will be fixed for you. Just feel free to contribute and promote your awesome work here! ü§© We'll get back to you in time ~ üòâ**


